File: PulseMonitor.Worker/Models/Metric.cs
namespace PulseMonitor.Worker.Models;

public class Metric
{
    public int Id { get; set; }
    public string DeviceId { get; set; }
    public double CpuUsage { get; set; }
    public double MemoryUsage { get; set; }
    public long Timestamp { get; set; }
}

File: PulseMonitor-UI/src/app/models/metric.ts
export interface Metric {
  deviceId: string;
  cpuUsage: number;
  memoryUsage: number;
  timestamp: number;
}

export const DEFAULT_METRIC: Metric = {
  deviceId: 'N/A',
  cpuUsage: 0,
  memoryUsage: 0,
  timestamp: 0,
};
